bulk-model-sync-mps/src/main/kotlin/org/modelix/mps/model/sync/bulk/MPSBulkSynchronizer.kt

```diff
@@ -66,33 +66,29 @@ object MPSBulkSynchronizer {
         val includedModuleNames = parseRawPropertySet(System.getProperty("modelix.mps.model.sync.bulk.input.modules"))
         val includedModulePrefixes = parseRawPropertySet(System.getProperty("modelix.mps.model.sync.bulk.input.modules.prefixes"))
         val inputPath = System.getProperty("modelix.mps.model.sync.bulk.input.path")
-        val access = repository.modelAccess
+        val jsonFiles = File(inputPath).listFiles()?.filter {
+            it.extension == "json" && isModuleIncluded(it.nameWithoutExtension, includedModuleNames, includedModulePrefixes)
+        }
 
-        access.runWriteInEDT {
-            val allModules = repository.modules
-            val includedModules: Iterable<SModule> = allModules.filter {
-                isModuleIncluded(it.moduleName!!, includedModuleNames, includedModulePrefixes)
-            }
-            val numIncludedModules = includedModules.count()
+        if (jsonFiles.isNullOrEmpty()) error("no json files found for included modules")
 
+        println("Found ${jsonFiles.size} modules to be imported")
+        val access = repository.modelAccess
+        access.runWriteInEDT {
             access.executeCommand {
-                for ((index, module) in includedModules.withIndex()) {
-                    println("Importing module ${index + 1} of $numIncludedModules: '${module.moduleName}'")
-                    val moduleFile = File(inputPath + File.separator + module.moduleName + ".json")
-                    if (moduleFile.exists()) {
-                        val importer = ModelImporter(
-                            MPSModuleAsNode(module),
-                        )
-                        importer.import(moduleFile)
-                    }
-                }
+                val repoAsNode = MPSRepositoryAsNode(repository)
```

Comment:
What benefits does this have over importing into `MPSModulesAsNode` in a for-loop?

Reply:
Running the import per module means, that the indices are limited to the scope of the module, which causes references across modules to not be resolved correctly.